network-api/networkapi/wagtailcustomization/templatetags/wagtailcustom_tags.py

```diff
@@ -0,0 +1,74 @@
+import re
+from django import template
+from django.utils.text import slugify
+from django.utils.html import strip_tags
+from wagtail.core.rich_text import RichText
+
+# We don't actually register any tags: the idea is to tap into
+# the richtext filter, but that won't let us change _all_ the
+# RichText content, so instead of setting up a filter override,
+# we literally reach into the RichText class. And change it.
+register = template.Library()
+
+# We'll be wrapping the original RichText.__html__(), so make
+# sure we have a reference to it that we can call.
+__original__html__ = RichText.__html__
+
+# This matches an h1/.../h6, using a regexp that is only
+# guaranteed to work because we know that the source of
+# the HTML code we'll be working with generates nice
+# and predictable HTML code.
+heading_re = r"<h(\d)[^>]*>([^<]*)</h\1>"
+
+
+def add_id_attribute(match):
+    """
+    This is a regexp replacement function that takes
+    in the above regex match results, and then turns:
+
+        <h1>some text</h1>
+
+    Into:
+
+        <h1><a class="fragment-id" id="some-text"></a>some text</h1>
+
+    where the id attribute value is generated by running
+    the heading text through Django's slugify() function.
+
+    This replacement will only kick in for h1/h2/h3.
+    """
+    n = match.group(1)
+    text_content = match.group(2)
+
+    if int(n) > 3:
+        return text_content
+
+    id = slugify(strip_tags(text_content))
+    return f'<h{n}><a class="fragment-id" id="{id}"></a>{text_content}</h{n}>'
```

Comment:
I'm still confused to the need for the anchor element here. Why can't we put the class and id on the heading element itself?

Reply:
because the anchor gets offset by 85px above the heading. Otherwise navigating to it puts the heading under the sticky nav bar.

Reply:
Duh. Right, it's been a while since we opened this PR. Since it's non-obvious, can you put that in a comment between 46 and 47?

Reply:
I've update the code comments.